cmd/collector/app/metrics.go

```diff
@@ -237,6 +249,9 @@ func (m *countsBySvc) countByServiceName(serviceName string, isDebug bool) {
 			debugStr = "true"
 		}
 		tags := map[string]string{"svc": serviceName, "debug": debugStr}
+		if samplerType != "" {
+			tags[samplerTypeKey] = samplerType
+		}
 		c := m.factory.Counter(metrics.Options{Name: m.category, Tags: tags})
 		counts[serviceName] = c
```

Comment:
we have a problem here where if a service has 2 sampler types (`lowerbound` and `probabilistic` both occur all the time together), we only create a counter once. We have to key this map differently so that we can store counters for multiple `sampler_type`s.

Reply:
what do you think if we just put `service_name + sampler_type` as key here.

Reply:
sure, but do something like `service_name + "$_$" + sampler_type` just in case there is a collision. Also, might want to use a string builder since string concatenation is expensive.

Reply:
Updated. Also I separate the `countsBySvc` for trace and span cause I'm using sync.Pool to reduce allocation of stringBuilder for trace whereas span metrics doesn't need that. Figured it made more sense to do so. Please let me know if you disagree.